src/Adapter/MSTest.TestAdapter/Execution/TestExecutionManager.cs

```diff
@@ -108,6 +108,9 @@ private static Task DefaultFactoryAsync(Func<Task> taskGetter)
     /// <param name="runContext">Context to use when executing the tests.</param>
     /// <param name="frameworkHandle">Handle to the framework to record results and to do framework operations.</param>
     /// <param name="runCancellationToken">Test run cancellation token.</param>
+#if DEBUG
+    [Obsolete("Use RunTestsAsync instead.")]
```

Comment:
So obsolete just for us, right?

Reply:
@Evangelink Yup. The whole class is obsolete publicly, but the obsoletion of it is "suppressed" for us. Now, we obsolete this one specifically for us in a way that doesn't get suppressed.